types/index.d.ts

```diff
@@ -16,73 +82,13 @@ declare module '@react-native-async-storage/async-storage' {
    *
    * @see https://react-native-async-storage.github.io/async-storage/docs/api
    */
-  export interface AsyncStorageStatic {
-    /**
-     * Fetches key and passes the result to callback, along with an Error if there is any.
-     */
-    getItem(key: string, callback?: (error?: Error, result?: string) => void): Promise<string | null>;
-
-    /**
-     * Sets value for key and calls callback on completion, along with an Error if there is any
-     */
-    setItem(key: string, value: string, callback?: (error?: Error) => void): Promise<void>;
-
-    removeItem(key: string, callback?: (error?: Error) => void): Promise<void>;
-
-    /**
-     * Merges existing value with input value, assuming they are stringified json. Returns a Promise object.
-     * Not supported by all native implementation
-     */
-    mergeItem(key: string, value: string, callback?: (error?: Error) => void): Promise<void>;
-
-    /**
-     * Erases all AsyncStorage for all clients, libraries, etc. You probably don't want to call this.
-     * Use removeItem or multiRemove to clear only your own keys instead.
-     */
-    clear(callback?: (error?: Error) => void): Promise<void>;
-
-    /**
-     * Gets all keys known to the app, for all callers, libraries, etc
-     */
-    getAllKeys(callback?: (error?: Error, keys?: string[]) => void): Promise<string[]>;
-
-    /**
-     * multiGet invokes callback with an array of key-value pair arrays that matches the input format of multiSet
-     */
-    multiGet(
-      keys: string[],
-      callback?: (errors?: Error[], result?: [string, string | null][]) => void
-    ): Promise<[string, string | null][]>;
-
-    /**
-     * multiSet and multiMerge take arrays of key-value array pairs that match the output of multiGet,
-     *
-     * multiSet([['k1', 'val1'], ['k2', 'val2']], cb);
-     */
-    multiSet(keyValuePairs: string[][], callback?: (errors?: Error[]) => void): Promise<void>;
-
-    /**
-     * Delete all the keys in the keys array.
-     */
-    multiRemove(keys: string[], callback?: (errors?: Error[]) => void): Promise<void>;
-
-    /**
-     * Merges existing values with input values, assuming they are stringified json.
-     * Returns a Promise object.
-     *
-     * Not supported by all native implementations.
-     */
-    multiMerge(keyValuePairs: string[][], callback?: (errors?: Error[]) => void): Promise<void>;
-  }
-
-  export function useAsyncStorage(key: string): {
-    getItem(callback?: (error?: Error, result?: string) => void): Promise<string | null>;
-    setItem(value: string, callback?: (error?: Error) => void): Promise<void>;
-    mergeItem(value: string, callback?: (error?: Error) => void): Promise<void>;
-    removeItem(callback?: (error?: Error) => void): Promise<void>;
-  }
-
-  const AsyncStorage: AsyncStorageStatic;
+  export function useAsyncStorage(key: string): AsyncStorageHook
+  const AsyncStorageLib: AsyncStorage;
+  export default AsyncStorageLib;
```

Comment:
It worked well in mine. Because this is default export, it does not matter. If you means that this seems like wrong naming convention, I'll fix it.

Reply:
It worked well for mine. Because this is default export, naming did not matter.
If you means it seems like wrong naming convention, I'll fix it.

Reply:
I was just wondering if we need this line at all: https://github.com/react-native-async-storage/async-storage/blob/ed50c7721978404c1532ea077f93f812745f2a05/types/index.d.ts#L86

Why can't it just be:
```ts
export default AsyncStorage;
```

Reply:
Because AsyncStorage itself is type, if we export it directly, It makes error like image.

<img width="735" alt="스크린샷 2021-11-24 오후 5 03 13" src="https://user-images.githubusercontent.com/16593102/143198339-81bfc38d-105d-4720-b41a-583a4bbc2e31.png">

Reply:
Ah, that makes sense. Thanks for educating me 😄